avocado/core/tree.py

```diff
@@ -200,6 +200,21 @@ def __ne__(self, other):
         """ Inverted eq """
         return not self == other
 
+    def __hash__(self):
+        values = []
+        for item in self.value:
+            try:
+                values.append(hash(item))
+            except TypeError:
+                values.append(hash(str(item)))
+        children = []
+        for item in self.children:
+            try:
+                children.append(hash(item))
+            except TypeError:
+                children.append(hash(str(item)))
+        return hash((self.name, ) + tuple(values) + tuple(children))
```

Comment:
Yes, this is a tricky one. I'd not come up with a better approach myself, but it's seems like this is good enough.